atter-logic/src/commonMain/kotlin/com.attendance.checker/events/viewModels/EventEditViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.attendance.checker.events.viewModels
+
+import com.attendance.checker.date.KDate
+import com.attendance.checker.events.Event
+import com.attendance.checker.events.SaveEvent
+import com.attendance.checker.events.presenters.EventEditPresenter
+import com.attendance.checker.shared.viewModels.ViewModel
+import com.attendance.checker.uuid.KUUID
+
+class EventEditViewModel constructor(private val presenter: EventEditPresenter,
+                                     private var newEventUseCase: SaveEvent,
+                                     private val event: Event? = null): ViewModel {
+
+    private var name: String?
+    private var date: KDate
+
+    companion object {
+        fun instance(presenter: EventEditPresenter, event: Event?) = EventEditViewModel(presenter, SaveEvent(), event)
+    }
+
+    init {
+        name = event?.name
+        date = event?.eventDate ?: KDate.now()
+    }
+
+    override fun onStart() {
+        refreshSaveButtonStatus()
+        presenter.displayName(name)
+        presenter.displayDate(date.isoDate)
+    }
+
+    fun didChangeName(name: String) {
```

Comment:
I'd recommend to move this logic to property setter 
```
var date : KDate
   set(value) { 
       field = value
       refreshSaveButtonStatus()
  }
```

Reply:
Good point.